### PR TITLE
posix: implement clock_getres()

### LINKED ISSUE #59955: posix: implement `clock_getres()`
Blocked for now, see: https://github.com/zephyrproject-rtos/zephyr/pull/60187#pullrequestreview-1521704680

> This function will need to tie in to specific hardware. Part of the LTSv3 Roadmap is to rework clocks, add device tree chosen nodes for RTC and monotonic, and provide a means of querying the precision of the HW clock resolution.
> 
> All of that should be done in Zephyr so that the POSIX layer is just a thin wrapper.
> 
> Clock getres should likely come after all of that is done..

There are currently 20 missing POSIX functions with trivial implementations, `clock_getres()` is one of them.

> clock_getres, clock_gettime, clock_settime - clock and timer functions
> 
> The clock_getres() function shall return the resolution of any clock. Clock resolutions are implementation-defined and cannot be set by a process. If the argument res is not NULL, the resolution of the specified clock shall be stored in the location pointed to by res. If res is NULL, the clock resolution is not returned. If the time argument of clock_settime() is not a multiple of res, then the value is truncated to a multiple of res.
> 
> The clock_gettime() function shall return the current value tp for the specified clock, clock_id.
> 
> The clock_settime() function shall set the specified clock, clock_id, to the value specified by tp. Time values that are between two consecutive non-negative integer multiples of the resolution of the specified clock shall be truncated down to the smaller multiple of the resolution.
> ...

See https://pubs.opengroup.org/onlinepubs/9699919799/functions/clock_getres.html

See RFC #51211 for more info.

### PR DESCRIPTION
**Pull request by harshilbhatt2001**
Address missing POSIX functions and test #59955

### CODE REVIEW

File: tests/posix/common/src/clock.c [hunk 1]
```diff
@@ -36,6 +37,12 @@ ZTEST(posix_apis, test_clock)
 		secs_elapsed = (te.tv_sec - ts.tv_sec - 1);
 	}
 
+	ret = clock_getres(CLOCK_MONOTONIC, &te);
+	zassert_equal(ret, 0, "Fail to get monotonic clock resolution");
```
Comment (ycsin) [inline tests/posix/common/src/clock.c:41]: can use `zassert_ok` to avoid declaring `ret`

File: lib/posix/clock.c [hunk 2]
```diff
@@ -106,6 +106,41 @@ int clock_settime(clockid_t clock_id, const struct timespec *tp)
...
+
+	case CLOCK_REALTIME:
+		uint64_t tick_1 = k_uptime_ticks();
+		uint64_t tick_2 = tick_1;
+
+		while (tick_1 == tick_2) {
+			tick_2 = k_uptime_ticks();
+		}
+
+		int tick_res = tick_2 - tick_1;
```
Comment (ycsin) [inline lib/posix/clock.c:130]: `tv_nsec` is in `long`, this prob can use `long` as well?

**Review by ycsin — COMMENTED** [review]
triggered the PR to rerun the tests

**Review by cfriedt — CHANGES_REQUESTED** [review]
This function will need to tie in to specific hardware. Part of the LTSv3 Roadmap is to rework clocks, add device tree chosen nodes for RTC and monotonic, and provide a means of querying the precision of the HW clock resolution.

All of that should be done in Zephyr so that the POSIX layer is just a thin wrapper. 

Clock getres should likely come after all of that is done..

**Comment by harshilbhatt2001** [comment]
Should this be closed, then?

**Comment by cfriedt** [comment]
Probably for now, but I appreciate the effort.

**Pull request closed without merging**